src/nvim/misc1.c

```diff
@@ -2927,12 +2930,11 @@ char_u *vim_getenv(char_u *name, int *mustfree)
     return p;
   }
 
-  vimruntime = (STRCMP(name, "VIMRUNTIME") == 0);
```

Comment:
Indeed, I've never seen environment variables interpreted case insensitively, and for good reason given how it doesn't adhere to the principle of least surprise, e.g. I'd be a little confused if `$ echo $VIMRUNTIME` showed nothing, but nvim was obeying `$ViMrUnTiMe`.

(I'm only mentioning this because I see some `STRICMP` calls still present in this PR)